File: Core/Entities/Appointment.cs
namespace Core.Entities
{
    public partial class Appointment
    {
        public int Id { get; set; }
        public int ClientId { get; set; }
        public AppUser Client { get; set; }
        public int? TermId { get; set; }
        public Term Term { get; set; }
        public int ServiceId { get; set; }
        public Service Service { get; set; }
        public DateTime ReservationDate { get; set; } = DateTime.Now;
        public bool IsCanceled { get; set; }      
    }
}

File: UI/barbershop_admin/lib/models/appointment.dart
import 'package:json_annotation/json_annotation.dart';

part 'appointment.g.dart';

@JsonSerializable()
class Appointment {
  int? id;
  int? clientId;
  String? clientName;
  int? termId;
  DateTime? date;
  String? startTime;
  String? endTime;
  int? serviceId;
  String? serviceName;
  double? servicePrice;
  DateTime? reservationDate;
  bool? isCanceled;

  Appointment({
    this.id,
    this.clientId,
    this.clientName,
    this.termId,
    this.date,
    this.startTime,
    this.endTime,
    this.serviceId,
    this.serviceName,
    this.servicePrice,
    this.reservationDate,
    this.isCanceled,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) =>
      _$AppointmentFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentToJson(this);
}